Server/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp

```diff
@@ -941,6 +946,47 @@ int CLuaPlayerDefs::TakePlayerScreenShot(lua_State* luaVM)
     return 1;
 }
 
+int CLuaPlayerDefs::SetPlayerScriptDebugLevel(lua_State* luaVM)
+{
+    CElement*    pElement;
+    unsigned int uiMode;
+    bool         bHideDebugger = false;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pElement);
+    if (argStream.NextIsBool())
+    {
+        bool bTemp;
+        argStream.ReadBool(bTemp);
+
+        if (bTemp)
+            argStream.SetCustomError("You can only pass false to hide the debug window or a level (0-3)");
+        else
+            bHideDebugger = true;
+    }
+    else
+    {
+        argStream.ReadNumber(uiMode);
+
+        if (uiMode < 0 || uiMode > 3)
+            argStream.SetCustomError("Invalid level (0-3)");
+    }
+
+    if (!argStream.HasErrors())
```

Comment:
As per #821 we should do a hard error here like so:
```cpp
if (argStream.HasErrors())
{
    return luaL_error(luaVM, argStream.GetFullErrorMessage());
}
```
and then move the passing code after that.